tfhe/src/integer/gpu/server_key/radix/abs.rs

```diff
@@ -0,0 +1,96 @@
+                    &d_bsk.d_vec,
+                    &self.key_switching_key.d_vec,
+                    self.message_modulus,
+                    self.carry_modulus,
+                    d_bsk.glwe_dimension,
+                    d_bsk.polynomial_size,
+                    self.key_switching_key
+                        .input_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key
+                        .output_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key.decomposition_level_count(),
+                    self.key_switching_key.decomposition_base_log(),
+                    d_bsk.decomp_level_count,
+                    d_bsk.decomp_base_log,
+                    num_blocks,
+                    PBSType::Classical,
+                    LweBskGroupingFactor(0),
+                );
+            }
+            CudaBootstrappingKey::MultiBit(d_multibit_bsk) => {
+                unchecked_signed_abs_radix_kb_assign_async(
+                    streams,
+                    &mut ct.as_mut().d_blocks.0.d_vec,
+                    &d_multibit_bsk.d_vec,
+                    &self.key_switching_key.d_vec,
+                    self.message_modulus,
+                    self.carry_modulus,
+                    d_multibit_bsk.glwe_dimension,
+                    d_multibit_bsk.polynomial_size,
+                    self.key_switching_key
+                        .input_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key
+                        .output_key_lwe_size()
+                        .to_lwe_dimension(),
+                    self.key_switching_key.decomposition_level_count(),
+                    self.key_switching_key.decomposition_base_log(),
+                    d_multibit_bsk.decomp_level_count,
+                    d_multibit_bsk.decomp_base_log,
+                    num_blocks,
+                    PBSType::MultiBit,
+                    d_multibit_bsk.grouping_factor,
+                );
+            }
+        };
+    }
+    pub fn unchecked_abs<T>(&self, ct: &T, streams: &CudaStreams) -> T
+    where
+        T: CudaIntegerRadixCiphertext,
+    {
+        let mut res = unsafe { ct.duplicate_async(streams) };
+        if T::IS_SIGNED {
+            unsafe { self.signed_unchecked_abs_assign_async(&mut res, streams) };
+            streams.synchronize();
+        }
+        res
+    }
+    pub fn abs<T>(&self, ct: &T, streams: &CudaStreams) -> T
+    where
+        T: CudaIntegerRadixCiphertext,
+    {
+        let mut res = unsafe { ct.duplicate_async(streams) };
+        if !ct.block_carries_are_empty() {
+            unsafe { self.full_propagate_assign_async(&mut res, streams) };
+        };
+        if T::IS_SIGNED {
+            unsafe { self.signed_unchecked_abs_assign_async(&mut res, streams) };
+            streams.synchronize();
+        }
```

Comment:
Here there should be an else with streams.synchronize() to be sure unsigned is correctly synchronized as well.